client/src/components/SearchBar.js

```diff
@@ -1,32 +1,35 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect, useRef } from 'react';
 import search from '../assets/search.svg';
-import { toHexColor } from '../util';
 
 function SearchBar(props) {
+    const scrollRef = useRef(null);
     const [ inputValue, setInputValue ] = useState("");
-    const [ query, setQuery ] = useState([]);
+
+    const handleScroll = () => {
+        const position = window.pageYOffset;
+        if (position < 300) {
+            scrollRef.current.className = 'search';
+        } else if (position >= 100) {
+            scrollRef.current.className = 'search fixed';
+        }
+    };
+
+    useEffect(() => {
+        window.addEventListener('scroll', handleScroll, { passive: true });
+    
+        return () => {
+            window.removeEventListener('scroll', handleScroll);
+        };
+    }, []);
 
     const handleOnChange = (e) => {
         // TODO: perhaps break at new tags? actually use query?
         setInputValue(e.target.value);
     }
 
-    const tags = query.map((tag, i) => {
```

Comment:
does it still filter without tags?

Reply:
This was back when i was ambitious lol

Reply:
Where doing something like `#tag<space>` had this cool look of an actual tag in the search bar and then you could add more tags :')

Reply:
ooo ok bet lgtm then!